pyaction/pyaction_wconprod_insert_kw.py

```diff
@@ -1,84 +1,55 @@
 # The python code in this file does the same as ACT-01 to ACT-05 of opm-tests/actionx/ACTIONX_WCONPROD.DATA
 import datetime
+import opm_embedded
 
-def run(ecl_state, schedule, report_step, summary_state, actionx_callback):
-    if (not 'act01_last_executed_at' in globals()):
-        globals()["act01_last_executed_at"] = datetime.datetime(1970, 1, 1) # Set act01_last_executed_at to the beginning of time
-        globals()["act02_last_executed_at"] = datetime.datetime(1970, 1, 1) # Set act02_last_executed_at to the beginning of time
-        globals()["act03_last_executed_at"] = datetime.datetime(1970, 1, 1) # Set act03_last_executed_at to the beginning of time
-        globals()["act01_execution_counter"] = 0
-        globals()["act02_execution_counter"] = 0
-        globals()["act03_execution_counter"] = 0
-        globals()["act04_executed"] = False
-        globals()["act05_executed"] = False
+ecl_state = opm_embedded.current_ecl_state
+summary_state = opm_embedded.current_summary_state
+schedule = opm_embedded.current_schedule
+report_step = opm_embedded.current_report_step
 
-    if (not "FU_GASFL" in summary_state):
-        summary_state["FU_GASFL"] = 0 # Gas Field Flag Set to Off
+if (not 'setup_done' in locals()):
+    last_updated_at = [datetime.datetime(1970, 1, 1)] * 3 # Set last_updated_at to the beginning of time
+    execution_counter = [0] * 3
+    converted_to_gas_field = False
+    all_wells_shut = False
+    setup_done = True
 
-    current_time = schedule.start + datetime.timedelta(seconds=summary_state.elapsed())
+current_time = schedule.start + datetime.timedelta(seconds=summary_state.elapsed())
 
-    act01_days_since_last_execution = (current_time - globals()["act01_last_executed_at"]).total_seconds()/3600/24  
-    act02_days_since_last_execution = (current_time - globals()["act02_last_executed_at"]).total_seconds()/3600/24  
-    act03_days_since_last_execution = (current_time - globals()["act03_last_executed_at"]).total_seconds()/3600/24  
+days_since_last_update = [(current_time - last_updated_at[0]).total_seconds()/3600/24, (current_time - last_updated_at[1]).total_seconds()/3600/24, (current_time - last_updated_at[2]).total_seconds()/3600/24]
```

Comment:
Newbee question :sweat_smile:: Would this also work?
```suggestion
days_since_last_update = [(current_time - i).total_seconds()/3600/24 for i in last_updated_at]
```